### DISCUSSION TITLE
Specifying filename for download

### DISCUSSION THREAD

**Discussion by jim-steenburgh** [blaylockbk/Herbie#58] [General]
Is there a way to specify the filename to write to when doing H.download?

**Comment by blaylockbk** [comment]
That is a very good question. I originally designed Herbie to mimic the directory structure on the remote. But it does look like you can change the filename with a little hack.

When you create a Herbie object, there is an attribute `LOCALFILE` that you can modify before you download.

```python
from herbie.archive import Herbie
H = Herbie('2021-01-01', model='hrrr', product='sfc', fxx=6)
H.LOCALFILE = 'custom_name'  # <-- Name the file whatever you want
H.download()
``` 
```
✅ Success! Downloaded HRRR from aws                 
	src: https://noaa-hrrr-bdp-pds.s3.amazonaws.com/hrrr.20210101/conus/hrrr.t00z.wrfsfcf06.grib2
	dst: C:\Users\blayl_depgywe\data\hrrr\20210101\cutsom_name
```

You can use the Herbie attributes to be more specific with the custom filename:
```python
H.LOCALFILE = f"model_{H.model.upper()}_{H.date:%Y%m%d%H}_F{H.fxx:02d}_valid_{H.valid_date:%Y%m%d%H}.grib2"
H.download()
# Downloads file: model_HRRR_2021010100_F06_valid_2021010106.grib2

H.LOCALFILE = f"model_{H.model.upper()}_surfaceFields_{H.date:%Y%m%d%H}_F{H.fxx:02d}.grib2"
H.download()
# Downloads file: model_HRRR_surfaceFields_2021010100_F06.grib2
```
Caveats: 
- **This method only works when downloading full files** and not subsets, because I use a hashed name to force a unique filename when downloading subsets. 
- You can't change the fact that the Herbie will put the file in the `hrrr/YYYYMMDD` directory. _(note to self, this is because the directory is hardcoded in the `get_localFilePath` method.)_
 
I created a new issue https://github.com/blaylockbk/Herbie/issues/60 and I'll see if I can implement a more intuitive way to do this.
